addon/ng2/commands/serve.ts

```diff
@@ -0,0 +1,129 @@
+    { name: 'host',                 type: String,                          aliases: ['H'],     description: 'Listens on all interfaces by default' },
+    { name: 'proxy',                type: String,                          aliases: ['pr', 'pxy'] },
+    { name: 'insecure-proxy',       type: Boolean, default: false,         aliases: ['inspr'], description: 'Set false to proxy self-signed SSL certificates' },
+    { name: 'watcher',              type: String,  default: 'events',      aliases: ['w'] },
+    { name: 'live-reload',          type: Boolean, default: true,          aliases: ['lr'] },
+    { name: 'live-reload-host',     type: String,                          aliases: ['lrh'],   description: 'Defaults to host' },
+    { name: 'live-reload-base-url', type: String,                          aliases: ['lrbu'],  description: 'Defaults to baseURL' },
+    { name: 'live-reload-port',     type: Number,                          aliases: ['lrp'],   description: '(Defaults to port number within [49152...65535])' },
+    { name: 'live-reload-live-css', type: Boolean, default: true,                              description: 'Whether to live reload CSS (default true)' },
+    { name: 'environment',          type: String,  default: 'development', aliases: ['e', { 'dev': 'development' }, { 'mat': 'material'},  { 'prod': 'production' }] },
+    { name: 'output-path',          type: 'Path',  default: 'dist/',       aliases: ['op', 'out'] },
+    { name: 'ssl',                  type: Boolean, default: false },
+    { name: 'ssl-key',              type: String,  default: 'ssl/server.key' },
+    { name: 'ssl-cert',             type: String,  default: 'ssl/server.crt' }
+  ],
+
+  run: function(commandOptions: ServeTaskOptions) {
+
+
+    commandOptions.liveReloadHost = commandOptions.liveReloadHost || commandOptions.host;
+
+    return this._checkExpressPort(commandOptions)
+      .then(this._autoFindLiveReloadPort.bind(this))
+      .then(function(commandOptions: ServeTaskOptions) {
+        commandOptions = assign({}, commandOptions, {
+          baseURL: this.project.config(commandOptions.environment).baseURL || '/'
+        });
+
+        if (commandOptions.proxy) {
+          if (!commandOptions.proxy.match(/^(http:|https:)/)) {
+            var message = 'You need to include a protocol with the proxy URL.' + EOL + 'Try --proxy http://' + commandOptions.proxy;
+
+            return Promise.reject(new SilentError(message));
+          }
+        }
+
+        const ServeWebpackTask = (require('../tasks/serve-webpack.ts'))
+
+        var serve = new ServeWebpackTask({
+          ui: this.ui,
+          analytics: this.analytics,
+          project: this.project,
+        });
+
+        return win.checkWindowsElevation(this.ui).then(function() {
+          return serve.run(commandOptions);
+        });
+      }.bind(this));
+  },
+
+  _checkExpressPort: function(commandOptions: ServeTaskOptions) {
```

Comment:
This is an object that is used as an argument for `Command.extend({})`, not a class. Maybe it can receive a class but none of the other commands do that. You sure you want that to be done?